src/DOMTokenList.js

```diff
@@ -27,7 +27,22 @@ module.exports = class DOMTokenList extends Array {
   }
 
   add(...tokens) {
-    this.splice(0, this.length, ...new Set(this.concat(tokens)));
+    const current = this.length;
+    const adding = tokens.length;
+    if (adding === 1) {
+      const token = tokens[0];
+      for (let i = 0; i < current; i++) {
+        if (this[i] === token) return;
+      }
+      this.push(token);
+    } else {
+      main: for (let i = 0; i < adding; i++) {
+        for (let j = 0; j < current; j++) {
```

Comment:
I cannot see anything in that link but O(n^2) wasn't a question, rather a statement. The algorithm used in here is not even a performance oriented one so I'm afraid this change won't land in this project